src/Mod/Path/PathTests/TestPathSlot.py

```diff
@@ -0,0 +1,264 @@
+# *                                                                         *
+# *   This program is free software; you can redistribute it and/or modify  *
+# *   it under the terms of the GNU Lesser General Public License (LGPL)    *
+# *   as published by the Free Software Foundation; either version 2 of     *
+# *   the License, or (at your option) any later version.                   *
+# *   for detail see the LICENCE text file.                                 *
+# *                                                                         *
+# *   This program is distributed in the hope that it will be useful,       *
+# *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+# *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+# *   GNU Library General Public License for more details.                  *
+# *                                                                         *
+# *   You should have received a copy of the GNU Library General Public     *
+# *   License along with this program; if not, write to the Free Software   *
+# *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  *
+# *   USA                                                                   *
+# *                                                                         *
+# ***************************************************************************
+
+import FreeCAD
+import Path
+import PathScripts.PathJob as PathJob
+import PathScripts.PathSlot as PathSlot
+from PathTests.PathTestUtils import PathTestBase
+
+
+class TestPathSlot(PathTestBase):
+    """Unit tests for the Slot operation."""
+
+    @staticmethod
+    def _setup_test_environment():
+        """_setup_test_environment()...
+        This method is upon instantiation of this test class.  Add code and objects here
+        that are needed for the duration of the test() methods in this class.
+        This method does not have access to the class `self` reference.
+        """
+        # doc = FreeCAD.open(FreeCAD.getHomePath() + 'Mod/Path/PathTests/test_slot_01.fcstd')
+        doc_title = "TestSlot"
+        doc = FreeCAD.newDocument(doc_title)
+
+        # This is a real world test to make sure none of the tool chain broke
+        box0 = doc.addObject('Part::Box', 'Box')
+        box0.Length = 25.0
+        box0.Width = 50.0
+        box0.Height = 10.0
+        box1 = doc.addObject('Part::Box', 'Box')
+        box1.Length = 5.0  # X
+        box1.Width = 15.0  # Y
+        box1.Height = 5.0  # Z
+        box1.Placement = FreeCAD.Placement(FreeCAD.Vector(5.0, 0.0, 5.0), FreeCAD.Rotation(FreeCAD.Vector(0,0,1), 0))
+        doc.recompute()
+        cut = doc.addObject('Part::Cut', 'Cut')
+        cut.Base = box0
+        cut.Tool = box1
+        doc.recompute()
+
+        # Create Job object
+        PathJob.Create('Job', [doc.Cut], None)
+        doc.recompute()
```

Comment:
It would be simpler to just add a test file and load that. Would also make the tests simpler since you don't have to search for your feature each time.

Reply:
I have not found a good answer for this one due to the challenges your suggestion introduces.  Yes, I do have a test file started and would like to use it to simplify things a bit.  The problem I run into is that when I modify the geometry of the test file I am working on, the edge and face numbering are disrupted, which in turn forces me to have to resynchronize many of the edge and face names used in `test##()` methods.  
. 
So, for now, I am preferring to search for each feature until I have a test file that will not need much maintenance or will not re-enumerate the edge and face names with future additions.

For example:
![Snip macro screenshot-149768](https://user-images.githubusercontent.com/47639332/108389133-fb6a8680-71d4-11eb-88e7-28ca0cfc324b.png)
This test file is coming along nicely, but when I make additions, the face and edge numbers get disrupted, causing the need to go back into the test module and edit edge and face name references.

My goal is to use a test file.  For now, I would like to get a basic unit test created and functional and merged so we have a starting point and some help with testing changes to the operation.

Reply:
wow - this is not a  test file, it's a torture file ;)

maybe use multiple files, it's easier to verify multiple simple shapes than one big one which includes everything. sliptonic has a new internet connection, so we're allowed to use more test files ;)

Reply:
>... maybe use multiple files, it's easier to verify multiple simple shapes than one big one which includes everything. ...

My thinking is to use one file, but it will contain multiple bodies to be used as bases for Jobs in this Slot unit test.  This way I think I could add another body with more targeted slots related to the attached image above without having to alter that particular body. 
.
Does this single-file, multi-body strategy seem okay?